src/iter.rs

```diff
@@ -31,39 +31,194 @@ impl<T> Iterator for BitIter<T>
                 let idx = self.prefix[1] | bit;
+                // Take corresponding usize from layer below
                 self.masks[0] = self.set.layer0(idx as usize);
+                // Prefix of the complete index
                 self.prefix[0] = idx << BITS;
                 continue;
             }
-
+            // Look at third level
             if self.masks[2] != 0 {
+                // Take first bit that isn't zero
                 let bit = self.masks[2].trailing_zeros();
+                // Remove it from masks
                 self.masks[2] &= !(1 << bit);
+                // Calculate index of the bit in second level
                 let idx = self.prefix[2] | bit;
+                // Take corresponding usize from layer below
                 self.masks[1] = self.set.layer1(idx as usize);
+                // Prefix of the index of the second level
                 self.prefix[1] = idx << BITS;
                 continue;
             }
-
+            // Look at the 4th and highest level
             if self.masks[3] != 0 {
+                // Take first bit that isn't zero
                 let bit = self.masks[3].trailing_zeros();
+                // Remove it from masks
                 self.masks[3] &= !(1 << bit);
+                // Take corresponding usize from layer below
                 self.masks[2] = self.set.layer2(bit as usize);
+                // Prefix of the index of the third level
                 self.prefix[2] = bit << BITS;
                 continue;
             }
-
+            // There is no set indices left
             return None;
         }
     }
 }
+
+#[cfg(feature="parallel")]
+pub use self::par_iter::*;
+/// A `ParallelIterator` over a [`BitSetLike`] structure.
+///
+/// [`BitSetLike`]: ../trait.BitSetLike.html
+#[cfg(feature="parallel")]
+pub struct BitParIter<T>(T);
+#[cfg(feature="parallel")]
+mod par_iter {
+    use rayon::iter::ParallelIterator;
+    use rayon::iter::internal::{UnindexedProducer, UnindexedConsumer, Folder, bridge_unindexed};
+    use super::*;
+
+    #[cfg(target_pointer_width="64")]
+    pub const BITS_IN_USIZE: usize = 64;
+    #[cfg(target_pointer_width="32")]
+    pub const BITS_IN_USIZE: usize = 32;
+
+    impl<T> BitParIter<T> {
+        /// Creates a new `BitParIter`. You usually don't call this function
+        /// but just [`.par_iter()`] on a bit set.
+        ///
+        /// [`.par_iter()`]: ../trait.BitSetLike.html#method.par_iter
+        pub fn new(set: T) -> Self {
+            BitParIter(set)
+        }
+    }
+
+    impl<T> ParallelIterator for BitParIter<T>
+        where T: BitSetLike + Send + Sync,
+    {
+        type Item = Index;
+
+        fn drive_unindexed<C>(self, consumer: C) -> C::Result
+            where C: UnindexedConsumer<Self::Item>
+        {
+            bridge_unindexed(BitProducer((&self.0).iter()), consumer)
+        }
+    }
+
+    pub struct BitProducer<'a, T: 'a + Send + Sync>(pub BitIter<&'a T>);
```

Comment:
Without inspecting it in detail, should this be re-exported in `lib.rs`? Or does it really stay private-only? If so, you could maybe no mark it as `pub`.